### PR TITLE
Project: Testing Practice

### PR DESCRIPTION
**Pull request by taiwocoker**
Dear TSE,

We wrote modules for analyze, CeaserSipher, Calculator, reverse string, and Capitalize functions and wrote tests for each of them using jest. 
Thanks for reviewing.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -3,10 +3,7 @@
 > This is part of the To-do-list Project in [The Odin Project's](https://www.theodinproject.com/courses/javascript/lessons/testing-practice?ref=lnav) Javascript Curriculum.
 
 
-![screenshot]()
```
Comment (uryelah) [inline README.md:6]: - [x] Please add instructions on how to run the code and how to run the tests locally, that inclues all the necessary setup, instalations, and terminal commands.

File: src/analyse.js [hunk 2]
```diff
@@ -0,0 +1,9 @@
+function analyse(array) {
+  return {
+    average: array.reduce((a, b) => a + b, 0) / array.length,
```
Comment (uryelah) [inline src/analyse.js]: - [x] [OPTIONAL] What happens if the length of the array is 0? Can you handle that error?

File: src/analyse.js [hunk 3]
```diff
@@ -0,0 +1,9 @@
+function analyse(array) {
```
Comment (uryelah) [inline src/analyse.js]: - [x] Let's use ES6 arrow function expressions whenever possible instead of the `function` keyword.
Ex:
```
const analyse = array => {
...
}
```

File: src/analyse.test.js [hunk 4]
```diff
@@ -0,0 +1,17 @@
+const analyse = require('./analyse');
+
+test('should have average', () => {
```
Comment (uryelah) [inline src/analyse.test.js]: - [x] [OPTIONAL] You could add a test also for empty arrays or for when the type of the argument is not array.

File: src/calculator.js [hunk 5]
```diff
@@ -0,0 +1,38 @@
+/*
+  eslint-disable func-names,
+*/
+
+function Calculator(a, b) {
```
Comment (uryelah) [inline src/calculator.js]: - [x] Prefer arrow functions (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Arrow_functions) for this and it's methods.

File: src/capitalize.js [hunk 6]
```diff
@@ -0,0 +1,13 @@
+const capitalize = (string) => {
+  if (typeof (string) === 'string') {
+    const arr = string.split('');
```
Comment (uryelah) [inline src/capitalize.js]: - [x] [OPTIONAL] No need to split the string into an array. You can just get the first element using brackets as you would in arrays and concatenate it with .substring.

More:
https://www.w3schools.com/jsref/jsref_substring.asp

File: src/capitalize.test.js [hunk 7]
```diff
@@ -0,0 +1,9 @@
+const capitalize = require('./capitalize');
+
+test('capitalize bread to return Bread', () => {
+  expect(capitalize('bread')).toBe('Bread');
```
Comment (uryelah) [inline src/capitalize.test.js]: - [x] [OPTIONAL] It would be a good idea to test this with more than one valid string. Otherwise you can't really be sure it will work properly in every string. Try to have more than one test case.

File: src/cs_encryption.js [hunk 8]
```diff
@@ -0,0 +1,46 @@
+/*
+  eslint-disable func-names
+*/
+
+function CeasarSipher(string, shift) {
+  const alphabet = ['a', 'b', 'c', 'd', 'e', 'f', 'g', 'h', 'i',
```
Comment (uryelah) [inline src/cs_encryption.js:6]: - [ ] [OPTIONAL] If you store the alphabet inside an object with the letters as keys and the indexes as values, ex: `a: 0`, the look up time would be faster. 

Using array.include and array.indexOf can slow down your code.

File: src/cs_encryption.js [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+    'j', 'k', 'l', 'm', 'n', 'o', 'p', 'q', 'r',
+    's', 't', 'u', 'v', 'w', 'x', 'y', 'z'];
+
+  const encrypt = function () {
+    let res = '';
+    const arr = string.split('');
```
Comment (uryelah) [inline src/cs_encryption.js]: - [x] [OPTIONAL] No need to turn the string into an array, you can loop a string using its iterator with the `for of` loop.

More: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...of

File: src/reverse_string.test.js [hunk 10]
```diff
@@ -0,0 +1,9 @@
+const reverseString = require('./reverse_string');
+
+test('reverseString hello to return olleh', () => {
```
Comment (uryelah) [inline src/reverse_string.test.js]: - [x] [OPTIONAL] Remeber to add more than one valid test case.

**Review by uryelah — CHANGES_REQUESTED** [review]
Hi **Patricia Manezeu & Taiwo Coker**, I'm Sarah your reviewer for this PR 👩‍🚀 

You did a great job with this project! I loved how you also tested for errors!
There are just a couple of issues to fix and then you'll be free to write more awesome tests in your next projects.

# Required Changes 🏗️ 🐛 💡

- [x] Please complete the README instructions for running tests.

- [x] [OPTIONAL] I strongly recommend Please put the tests in a specific spec or tests folder. they should not be in the same folder as the functions.

(Check the comments below too)

-----

 **Kindly fix the pointed out issues and comments below before requesting the next code review**

------------

Cheers and Happy coding!👏👏👏

Code review by **Sarah Uryelah Silva Chamorro**

(Feel free to contact me on slack by the handle of  **sarah** if you have any additional question)

**Comment by patriciachrysy** [comment]
Applied all of the mandatory requested changes and some of the optional ones. Thanks for reviewing again.

**Review by rOluochKe — APPROVED** [review]
### You got it right, well done :clap: The project has been approved. Congratulations :boom: :100:
### Merge and submit the project. Good luck with the next one.
- Happy hacking!
